core/src/main/java/org/apache/calcite/rel/rules/AbstractMaterializedViewRule.java

```diff
@@ -1502,9 +1502,10 @@ protected MaterializedViewAggregateRule(RelOptRuleOperand operand,
               rexBuilder.makeInputRef(result,
                   groupSet.indexOf(inverseMapping.getTarget(i))));
         }
-        for (int i = 0; i < queryAggregate.getAggCallList().size(); i++) {
+        // We add aggregate functions that are present in result to projection list
+        for (int i = queryAggregate.getGroupCount(); i < result.getRowType().getFieldCount(); i++) {
           projects.add(
-              rexBuilder.makeInputRef(result, queryAggregate.getGroupCount() + i));
+              rexBuilder.makeInputRef(result, i));
         }
         result = relBuilder
```

Comment:
Could you please add a test?

Reply:
Thanks for the comment @chunweilei . We have not been able to reproduce this issue with a test case, please see JIRA for more background.